lib/spack/spack/spec.py

```diff
@@ -3171,7 +3206,7 @@ def __missing__(self, key):
 #
 # These are possible token types in the spec grammar.
 #
-HASH, DEP, AT, COLON, COMMA, ON, OFF, PCT, EQ, ID, VAL = range(11)
+HASH, FULL_HASH, DEP, AT, COLON, COMMA, ON, OFF, PCT, EQ, ID, VAL = range(12)
```

Comment:
There's no reason to add this here, since the lexer doesn't actually have a sigil for parsing full hashes.  I think it's totally reasonable to add print logic for the full hash.  But I would not add parse logic for it yet.  We are not transitioning to using full hashes to identify specs; we're just getting the package hashing in so that we can use it in the build farm.